core/container.go

```diff
@@ -339,6 +339,24 @@ func (container *Container) WithFS(ctx context.Context, dir *Directory) (*Contai
 	return &Container{ID: id}, nil
 }
 
+func (container *Container) WithDirectory(ctx context.Context, gw bkgw.Client, subdir string, src *Directory, filter CopyFilter) (*Container, error) {
+	return container.updateRootFS(ctx, gw, subdir, func(dir *Directory) (*Directory, error) {
+		return dir.WithDirectory(ctx, "", src, filter)
```

Comment:
These empty strings feel spooky to me for some reason, this feels more clear:

```suggestion
		return dir.WithDirectory(ctx, ".", src, filter)
```